TaskLayer/SearchTask/SearchTask.cs

```diff
@@ -453,5 +454,74 @@ private static MassDiffAcceptor ParseSearchMode(string text)
 
             return massDiffAcceptor;
         }
+
+        /// <summary>
+        /// Handle ambiguity when two theoretical proteins in the 
+        /// search space have the same accession number.
+        /// The accession must be unique for indexed searches
+        /// </summary>
+        /// <param name="proteins"></param>
+        /// <param name="tcAmbiguity"></param>
+        protected static void SanitizeProteinDatabase(List<Protein> proteins, TargetContaminantAmbiguity tcAmbiguity)
+        {
+            if (proteins.Count != 0) //if we have proteins
+            {
+                proteins.Sort((x, y) => x.Accession.CompareTo(y.Accession));
+                string previousAccession = proteins[0].Accession; //get first accession
+                for (int i = 1; i < proteins.Count; i++)
+                {
+                    string currentAccession = proteins[i].Accession; //get current accession
+                    if (currentAccession.Equals(previousAccession)) //if this accession was found in the previous protein (i.e. it's a duplicate)
```

Comment:
We want to compare accessions and not protein objects. The index will crash if two identical accessions are present. The mods, sequences, variants, etc. are irrelevant.

Reply:
But, it does matter which one gets deleted. If you have xml and fasta, then removing xml could remove peptides with PTMs not present in the FASTA. That's a non-zero consequence.

Reply:
Isn't the user determining which protein that they want to keep when they select whether they want to keep the contaminant or keep the decoy? Even if the contaminant version has mods not in the target database, if the user only wants to keep the target, then I am not sure it would be right for us to override that because we think one protein entry is better than the other.